File: domain/src/main/java/com/ahn/domain/currency/usecase/SaveCalculatorSelectionUseCase.kt
package com.ahn.domain.currency.usecase

import com.ahn.domain.currency.repository.CurrencySelectionRepository
import javax.inject.Inject

class SaveCalculatorSelectionUseCase @Inject constructor(private val repository: CurrencySelectionRepository) {
    suspend operator fun invoke(mainCode: String, subCode: String) {
        repository.saveCalculatorSelection(mainCode, subCode)
    }
}

File: ios/Packages/Domain/Sources/Domain/Currency/UseCase/SaveCalculatorSelectionUseCase.swift
public struct SaveCalculatorSelectionUseCase: Sendable {
    private let repository: any CurrencySelectionRepository

    public init(repository: any CurrencySelectionRepository) {
        self.repository = repository
    }

    public func callAsFunction(mainCode: String, subCode: String) async throws {
        try await repository.saveCalculatorSelection(mainCode: mainCode, subCode: subCode)
    }
}
